s31175-PiotrTrybala/lab3-rozwiazanie.txt

```diff
@@ -0,0 +1,261 @@
+0 (rozgrzewkowe)
+1. Przejdź do swojego katalogu domowego
+
+	cd $HOME
+
+2. Wydaj komendę: ls -a
+
+3. Zobacz ile plików wypisało.
+
+	Wypisało 22 pliki
+
+4. Teraz wydaj komendę: ls -a | grep D
+HINT: Jeśli nie masz żadnego pliku czy katalogu z literką D to wybierz inną literkę,
+która już będzie w nazwie jakiegoś pliku czy katalogu.
+
+	ls -a | grep l
+
+5. Zobacz ile teraz jest wyników. Co się stało?
+
+	Wypisane zostało 11 plików, każdy z nich zawiera w nazwie literę "l"
+
+Otóż program grep służy do wyszukiwania wierszy w pliku lub strumieniu wejściowym,
+które pasują do wzorca. Tu podano wzorzec jako "D".
+
+6. A teraz wykonaj komendę: ls -a | grep D > lista_plikow_z_literka_d.txt
+
+	ls -a | grep l > lista_plikow_z_literka_l.txt
+
+7. Zobacz czy utworzył sie jakiś plik?
+
+	Powstał plik "lista_plikow_z_literka_l.txt"
+
+Jaka jest jego treść?
+
+	Zawiera on wcześniej wypisaną listę
+
+Co znaczy | oraz co znaczy > ?
+
+	| to potok, przekierowuje on standardowe wyjście poprzedzającej go komendy
+	jako wejście komendy, która po nim następuje
+	> znak większości przekierowuje standardowe wyjście do pliku
+
+8. A teraz wykonaj komendę: ls -a | grep D | tee lista_plikow_z_literka_d_2.txt
+
+	ls -a | grep l | tee lista_plikow_z_literka_l_2.txt
+
+9. Zobacz czy plik się utworzył?
+
+	Tak, powstał nowy plik
+
+Jaka jest treść względem poprzedniej próby bez programu tee?
+
+	Treść pliku zapisanego przy pomocy programu "tee" nie zawiera pliku,
+	w którym została zapisana
+
+Co robi program tee?
+
+	Program tee jednocześnie wypisuje do konsoli oraz zapisuje do pliku standardowe wyjście programu
+
+1 Zadania:
+
+A: Zobacz co się stanie jeśli wpiszemy w terminalu:
+
+ps
+
+	Wyświetlona została lista procesów
+
+ps -a
+
+	Wyświetlona została lista wszystkich procesów poza 2-ma procesami sesji
+
+ps x
+
+	Wyświetlona została lista wszystkich procesów aktualnego użytkownika
+
+ps axu
+INFO: tu kolejność parametrów a, x, u nie ma znaczenie a jedynie fakt, że są podane.
+Dowolna kolejność powinna dać ten sam wynik.
+
+	Wyświetlona została lista wszystkich procesów w formie "przyjaznej" dla użytkownika
+
+Jak myślisz, co oznacza znak zapytania w kolumnie 2?
+
+	Znak zapytanie w kolumnie "TTY" oznacza brak przypisanego do procesu terminala,
+	najprawdopodobniej są to procesy uruchomione na serwerze
+
+Nie wiesz? A gdzie może być wyjaśnienie?
+
+	man ps
+
+B: Jak wykonasz poniższe 2 zadania?
+
+Wyświetl wszystkie procesy bash
+
+	ps ax | grep bash
```

Comment:
Kombinowanie...
`man ps` tam jest odpowiedź na to pytanie. Czyli jak szukać po nazwie procesu a nie po frazie w wierszu.